### PR TITLE
guile 3.0.2

### PR DESCRIPTION
**Pull request by alebcay**
Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by alebcay** [comment]
Looks like `autogen` is not detecting this version of `guile`. `mdk` detects and builds with this update without any issues.

```
configure: checking for guile 2.2
configure: checking for guile 2.0
configure: checking for guile 1.8
configure: error: 
No Guile development packages were found.

Please verify that you have Guile installed.  If you installed Guile
from a binary distribution, please verify that you have also installed
the development packages.  If you installed it yourself, you might need
to adjust your PKG_CONFIG_PATH; see the pkg-config man page for more.
```

**Comment by Bo98** [comment]
The 2.2 branch is still supported so a versioned formula is a possibility in the short term.

File: Formula/guile@2.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  sha256 "cdf776ea5f29430b1258209630555beea6d2be5481f9da4d64986b077ff37504"
+
+  head do
+    url "https://git.savannah.gnu.org/git/guile.git"
+
+    depends_on "autoconf" => :build
+    depends_on "automake" => :build
+    depends_on "gettext" => :build
+    uses_from_macos "flex" => :build
+  end
```
Comment (Bo98) [inline Formula/guile@2.rb:15]:
```suggestion
  keg_only :versioned_formula
```

File: Formula/guile@2.rb [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  depends_on "libunistring"
+  depends_on "pkg-config" # guile-config is a wrapper around pkg-config.
+  depends_on "readline"
+
+  def install
+    # Work around Xcode 11 clang bug
+    # https://bitbucket.org/multicoreware/x265/issues/514/wrong-code-generated-on-macos-1015
+    ENV.append_to_cflags "-fno-stack-check" if DevelopmentTools.clang_build_version >= 1010
+
+    system "./autogen.sh" unless build.stable?
```
Comment (Bo98) [inline Formula/guile@2.rb:31]:
```suggestion
```

File: Formula/guile@2.rb [hunk 3]
```diff
@@ -0,0 +1,58 @@
+class GuileAT2 < Formula
```
Comment (SMillerDev) [inline Formula/guile@2.rb:1]: This seems like a possible candidate to test the new depreciation DSL if we have that info.

Comment (alebcay) [reply Formula/guile@2.rb:1]: What information is needed for this?

Comment (SMillerDev) [reply Formula/guile@2.rb:1]: I didn't look at the DSL yet. But I thought it needed a date. @MikeMcQuaid do you have an example of the deprecation DSL?

Comment (Bo98) [reply Formula/guile@2.rb:1]: Is it the case we deprecate _every_ versioned formula or what is the line here? I ask because Guile 2.2 is still supported.

Comment (SMillerDev) [reply Formula/guile@2.rb:1]: My suggestion would be adding a deprecation to every versioned formula with a known EOL date.

Comment (MikeMcQuaid) [reply Formula/guile@2.rb:1]: `deprecate! :date => "01-03-2020"` (or any date it can parse). The `:date` is optional. There's also `disable!`

Comment (Bo98) [reply Formula/guile@2.rb:1]: > any date it can parse

dd-mm-yyyy or mm-dd-yyyy?

Comment (SMillerDev) [reply Formula/guile@2.rb:1]: https://xkcd.com/1179/

Comment (MikeMcQuaid) [reply Formula/guile@2.rb:1]: @Bo98 Try it and see 😂.

**Comment by alebcay** [comment]
I've now added `deprecate!` to `guile@2`.

**Comment by Bo98** [comment]
The superenv shim needs copying from the guile formula to guile@2

`Aliases/guile@3.rb` should be `Aliases/guile@3`

**Comment by alebcay** [comment]
The shim reference issue has been fixed and the alias name has been corrected.

**Review by MikeMcQuaid — APPROVED** [review]

**Pull request closed without merging**